tests/unit/generate_detect_target_contour.py

```diff
@@ -0,0 +1,226 @@
+    background: np.ndarray, landing_data: LandingPadImageConfig
+) -> NumpyImage:
+    """
+    Blurs an image and adds a singular landing pad to the background.
+
+    background: A numpy image.
+    landing_data: Landing pad data for the landing pad to be blurred and added.
+
+    Returns: Image with the landing pad.
+    """
+    x, y = background.shape[:2]
+
+    mask = np.zeros((x, y), np.uint8)
+    mask = cv2.ellipse(
+        mask,
+        landing_data.centre,
+        landing_data.axis,
+        landing_data.angle,
+        0,
+        360,
+        255,
+        -1,
+    )
+
+    mask = cv2.blur(mask, (25, 25), 7)
+
+    alpha = mask[:, :, np.newaxis] / 255.0
+    # Brings the image back to its original colour.
+    fg = np.full(background.shape, LANDING_PAD_COLOUR_BLUE, dtype=np.uint8)
+
+    blended = (background * (1 - alpha) + fg * alpha).astype(np.uint8)
+    return NumpyImage(blended)
+
+
+def draw_landing_pad(
+    image: np.ndarray, landing_data: LandingPadImageConfig
+) -> tuple[NumpyImage, BoundingBox]:
+    """
+    Draws a single landing pad on the provided image and saves the bounding box coordinates to a text file.
+
+    image: The image to add a landing pad to.
+    landing_data: Landing pad data for the landing pad to be added.
+
+    Returns: Image with landing pad and the bounding box for the drawn landing pad.
+    """
+    centre_x, centre_y = landing_data.centre
+    axis_x, axis_y = landing_data.axis
+    angle_in_rad = math.radians(landing_data.angle)
+
+    ux = axis_x * math.cos(angle_in_rad)
+    uy = axis_x * math.sin(angle_in_rad)
+
+    vx = axis_y * math.sin(angle_in_rad)
+    vy = axis_y * math.cos(angle_in_rad)
+
+    width = 2 * math.sqrt(ux**2 + vx**2)
+    height = 2 * math.sqrt(uy**2 + vy**2)
+
+    top_left = (int(max(centre_x - (0.5) * width, 0)), int(max(centre_y - (0.5) * height, 0)))
+    bottom_right = (
+        min(centre_x + (0.5) * width, image.shape[1]),
+        min(centre_y + (0.5) * height, image.shape[0]),
+    )
+
+    bounding_box = BoundingBox(top_left, bottom_right)
+
+    if landing_data.blur:
+        image = add_blurred_landing_pad(image, landing_data)
+        return image, bounding_box
+
+    image = cv2.ellipse(
+        image,
+        landing_data.centre,
+        landing_data.axis,
+        landing_data.angle,
+        0,
+        360,
+        LANDING_PAD_COLOUR_BLUE,
+        -1,
+    )
+    return NumpyImage(image), bounding_box
+
+
+def create_test(landing_list: list[LandingPadImageConfig]) -> InputImageAndExpectedBoundingBoxes:
+    """
+    Generates test cases given a data set.
+
+    landing_list: List of landing pad data to be generated.
+
+    Returns: The image and expected bounding box.
+    """
+    image = np.full(shape=(1000, 2000, 3), fill_value=255, dtype=np.int8)
+    confidence_and_label = [1, 0]
+
+    # List to hold the bounding boxes.
+    # boxes_list = [confidence, label, top_left_x, top_left_y, bottom_right_x, bottom_right_y]
```

Comment:
This is not true! `boxes_list[0]` is not confidence, `boxes_list[1]` is not label, etc.